source/nodejs/adaptivecards/src/schema.ts

```diff
@@ -27,17 +27,22 @@ export interface IBackgroundImage {
     url: string;    
 }
 export interface IAdaptiveCard extends ITypedElement {
-    version?: IVersion | string,
-    actions?: (IActionBase)[];
+    type: string;
+    version?: IVersion | string;
     backgroundImage?: IBackgroundImage | string;
-    body?: (ITextBlock | IImage | IImageSet | IFactSet | IContainer | IColumnSet | IInputDate | IInputNumber | IInputText | IInputTime | IInputToggle)[];
+    body?: (any)[];
+    actions?: (IActionBase)[];    
     speak?: string;
-    title?: string;
+    [propName: string]: any;
 }
 export interface ICardElement extends ITypedElement {
-    speak?: string;
+    type: string,
+    id?: string;
     horizontalAlignment?: HorizontalAlignment;
-    separation?: Separation;
+    spacing?: Separation;
+    separator?: boolean;
+    [propName: string]: any;
```

Comment:
Why this?

Reply:
It signifies that extra properties are allowed on the interface. Per [TypeScript docs](https://www.typescriptlang.org/docs/handbook/interfaces.html)

"However, a better approach might be to add a string index signature if you’re sure that the object can have some extra properties that are used in some special way."

Reply:
I get that, but why do we want this?

Reply:
Because adaptive cards/elements allow extra properties to be included

Reply:
Oh, I didn't realize we officially allowed that.

Reply:
Indeed :) they should be used very sparingly of course, but it's a supported construct. Let me know if you can approve the review